### PR TITLE
[KYUUBI #2438] Ranger AccessResource get database from spark catalog when database isnull

### LINKED ISSUE #2438: [Improvement] Shall We Add a SparkSession In AccessResource?
### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### Search before asking

- [X] I have searched in the [issues](https://github.com/apache/incubator-kyuubi/issues?q=is%3Aissue) and found no similar issues.

### What would you like to be improved?

The current constructor of AccessResource is :
```scala
object AccessResource {
  def apply(
      objectType: ObjectType,
      firstLevelResource: String,
      secondLevelResource: String,
      thirdLevelResource: String) {
    ...
  }
}
```
In some cases, firstLevelResource maybe null, for example, in spark 2.4,`DESCRIBE TABLE table_name`, will produce a `DescribeTableCommand` without database info, so the correspnding AccessResource's `firstLevelResource` is null. Then it will generate a invaild RangerRequest, because of the databasse field is null.

I'm not sure that when the database is not explicitly specified in the sql statement, there will be other commands without database information.

### How should we improve?

Therefore, I think we should add a sparksession in AccessResource, in the case of databasse is null. When the `firstLevelResource` is null or empty, get `firstLevelResource` from catalog's current database.

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### PR DESCRIPTION
**Pull request by packyan**
### _Why are the changes needed?_

To close #2438.
Since there are some commands without database information, when `firstLevelResource` is null or empty, it's reasonable to fallback to using the current db as `firstLevelResource`.

### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessResource.scala [hunk 1]
```diff
@@ -40,11 +41,20 @@ object AccessResource {
...
       secondLevelResource: String,
-      thirdLevelResource: String): AccessResource = {
+      thirdLevelResource: String,
+      spark: SparkSession = null): AccessResource = {
     val resource = new AccessResource(objectType)
 
+    val firstLevelResource =
+      if (spark != null
+        && (firstLevelResourceOrNull == null || firstLevelResourceOrNull.isEmpty)) {
+        spark.catalog.currentDatabase
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessResource.scala:53]: why session catalog？how about other catalog implementations

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessResourceSuite.scala [hunk 2]
```diff
@@ -57,3 +59,95 @@ class AccessResourceSuite extends AnyFunSuite {
...
+        val operationType = OperationType(plan.nodeName)
+        val tuple = PrivilegesBuilder.build(plan)
+
+        val resourceWithoutSparkSession = AccessResource(tuple._1.head, operationType)
+        assert(resourceWithoutSparkSession.objectType === TABLE)
+        assert(resourceWithoutSparkSession.getDatabase === dbName)
+        assert(resourceWithoutSparkSession.getTable equalsIgnoreCase tableName)
+
+        val resourceWithSparkSession = AccessResource(tuple._1.head, operationType, spark = spark)
+        assert(resourceWithSparkSession.objectType === TABLE)
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessResourceSuite.scala:134]: try view without database？

Comment (yaooqinn) [reply extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/AccessResourceSuite.scala:134]: desc ${viewName}

**Comment by yaooqinn** [comment]
I still think we shall handle the commands where the database is optional, and set currentDb accordingly, instead of setting it blindly in AccessResource

**Comment by packyan** [comment]
> 

Ok,  if we can handle the commands where the database is optional, it’s definitely better than setting it blindly in AccessResource. My solution may be too violent.

**Pull request closed without merging**